aten/src/TH/generic/THTensorRandom.cpp

```diff
@@ -427,24 +440,24 @@ void THTensor_(multinomial)(THLongTensor *self, THGenerator *_generator, THTenso
 #if defined(TH_REAL_IS_BYTE)
 void THTensor_(getRNGState)(THGenerator *_generator, THTensor *self)
 {
-  static const size_t size = sizeof(THGenerator);
-  THGenerator *rng_state;
+  static const size_t size = sizeof(THGeneratorState);
```

Comment:
I think we should lock the mutex in `getRNGState` and `setRNGState` to ensure the operations are effectively atomic even if they overlap with some random function on the same generator.